frontend/packages/kubevirt-plugin/src/models.ts

```diff
@@ -0,0 +1,136 @@
+import { K8sKind } from '@console/internal/module/k8s';
+
+export const VirtualMachineModel: K8sKind = {
+  label: 'Virtual Machine',
+  labelPlural: 'Virtual Machines',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachines',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachines',
+  abbr: 'VM',
+  namespaced: true,
+  kind: 'VirtualMachine',
+  id: 'virtualmachine',
+};
+
+export const VirtualMachineInstanceModel: K8sKind = {
+  label: 'Virtual Machine Instance',
+  labelPlural: 'Virtual Machine Instances',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachineinstances',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachineinstances',
+  abbr: 'VMI',
+  namespaced: true,
+  kind: 'VirtualMachineInstance',
+  id: 'virtualmachineinstance',
+};
+
+export const VirtualMachineInstancePresetModel: K8sKind = {
+  label: 'Virtual Machine Instance Preset',
+  labelPlural: 'Virtual Machine Instance Presets',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachineinstancepresets',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachineinstancepresets',
+  abbr: 'VMIP',
+  namespaced: true,
+  kind: 'VirtualMachineInstancePreset',
+  id: 'virtualmachineinstancepreset',
+};
+
+export const VirtualMachineInstanceReplicaSetModel: K8sKind = {
+  label: 'Virtual Machine Instance Replica Set',
+  labelPlural: 'Virtual Machine Instance Replica Sets',
+  apiVersion: 'v1alpha3',
+  path: 'virtualmachineinstancereplicaset',
+  apiGroup: 'kubevirt.io',
+  plural: 'virtualmachineinstancereplicasets',
+  abbr: 'VMIR',
+  namespaced: true,
+  kind: 'VirtualMachineInstanceReplicaSet',
+  id: 'virtualmachineinstancereplicaset',
+};
+
+/* TODO(mlibra): migrate templates
+export const VmTemplateModel: K8sKind = {
```

Comment:
This relates to #1658 - we really shouldn't redefine existing base k8s models.

I'm working on a PR that will integration-test actual Console plugins and detect any conflicts among their extensions.

(It's perfectly fine to do this in our Console fork, which is meant to support KubeVirt specifically.)